### PR TITLE
feat: add environment and commit_hash inputs

### LINKED ISSUE #15: Add support for default (main) branch and production deploys
**Is your feature request related to a problem? Please describe.**
I would like to perform actions like purging my Cloudflare cache and running Lighthouse tests when my default branch (`main`) is deployed, but the plugin doesn't currently support fetching data for production (non-preview) deployments.

**Describe the solution you'd like**
Github Action should be able to detect when a deploy for the default-branch has succeeded, and provide the url, even if it's a production (non-preview) deployment.

**Additional context**
I see that the current plugin behavior 
- filters based on `environment === "preview"`
- asserts against branch names (which makes things tricky when you have multiple deploys per branch)

At the same time, the Cloudflare API also provides a `deployment_trigger.metadata.commit_hash` property. 

Is it possible to allow the user to
- tell the plugin to match via commit hash instead of branch name, 
- tell the plugin which deploy environment to check (e.g., production)

### PR DESCRIPTION
**Pull request by ankushg**
Fix #15 

Adds inputs:
- `environment` (defaults to `"preview"`)
- `commit_hash` (not required)

Update deployment selection logic:
- filter by `commit_hash` when provided
- update `environment` selection to match input (instead of hardcoding to `"preview"`)

Update tests
- add a test to make sure we search through all deployment envs when `environment` is `null`
- add a test to make sure we use commitHash when given

### CODE REVIEW

File: test/cloudflare.test.js [hunk 1]
```diff
@@ -110,6 +266,13 @@ test('getDeploymentUrl() should fail if there are no matching builds', async ()
...
-    getDeploymentUrl('123xyz', 'zentered', 'fix/huge-bug', 'zentered.co')
+    getDeploymentUrl(
+      '123xyz',
+      'zentered',
+      'fix/huge-bug',
+      'zentered.co',
+      'preview',
+      null
+    )
   ).rejects.toThrow()
```
Comment (PatrickHeneise) [inline test/cloudflare.test.js:277]:
```suggestion
  ).rejects.toThrow('no matching builds found')
```

You discovered an issue with the tests here. Without `source` and `deployment_trigger`, there's another error that's being thrown, not the `"no matching builds"`. Could you change this test to include the missing attributes in the mock object?

File: cloudflare.mjs [hunk 2]
```diff
@@ -7,7 +7,9 @@ export default async function getDeploymentUrl(
   accountEmail,
   projectId,
   repo,
-  branch
+  branch,
+  environment,
```
Comment (PatrickHeneise) [inline cloudflare.mjs:11]: Is it worth introducing an `options` object for optional parameters? There may be more in the future.

File: cloudflare.mjs [hunk 3]
```diff
@@ -30,9 +32,14 @@ export default async function getDeploymentUrl(
...
   core.debug(`Looking for matching deployments ${repo}/${branch}`)
   const builds = data.result
-    .filter((d) => d.environment === 'preview')
     .filter((d) => d.source.config.repo_name === repo)
     .filter((d) => d.deployment_trigger.metadata.branch === branch)
+    .filter((d) => environment == null || d.environment === environment)
+    .filter(
+      (d) =>
+        commitHash == null ||
+        d.deployment_trigger.metadata?.commit_hash === commitHash
```
Comment (PatrickHeneise) [inline cloudflare.mjs:41]: JavaScript Actions still run on Node 12, the optional chaining doesn't work there. You can try to change the babel/ncc config to covert this, or refactor.

**Review by PatrickHeneise — CHANGES_REQUESTED** [review]
Looks good, thank you for the contribution! I left a few comments, would be great to get those fixed before merging.

**Comment by PatrickHeneise** [comment]
@ankushg what do you think about the changes?

File: cloudflare.mjs (hunk 2 above)
Comment (ankushg) [reply cloudflare.mjs:11]: Hm, I can see how that might be nice, but it might be annoying to add another level of nesting of optional things until we also get optional chaining. 

I'm not familiar enough with babel to know how to get that working…

**Review by PatrickHeneise — APPROVED** [review]

**Comment by PatrickHeneise** [comment]
Thanks @ankushg !
